cmd/juju/model/exportbundle.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+package model
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/api/bundle"
+	"github.com/juju/juju/cmd/modelcmd"
+)
+
+// NewexportbundleCommand returns a fully constructed exportbundle command.
+func NewExportBundleCommand() cmd.Command {
+	return modelcmd.Wrap(&exportBundleCommand{})
+}
+
+type exportBundleCommand struct {
+	modelcmd.ModelCommandBase
+	api ExportBundleModelAPI
+	// name of the charm bundle file.
+	Filename string
+}
+
+const exportBundleHelpDoc = `
+Exports the current model configuration into a YAML file.
+
+Examples:
+
+    juju export-bundle
+
+If --filename is not used, the archive is downloaded to a temp file
+which the name of the model with time appended.
+`
+
+// Info implements Command.
+func (c *exportBundleCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "export-bundle",
+		Purpose: "Exports the current model configuration in a charm bundle.",
+		Doc:     exportBundleHelpDoc,
+	}
+}
+
+// SetFlags implements Command.
+func (c *exportBundleCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	f.StringVar(&c.Filename, "filename", "", "Export Model")
```

Comment:
I am not sure that we'd want to limit users to only store the result in the file.

Generally, we provide both options - output into stdout or file. Also, if we do not lock users into a format on the server side, we can be really user-friendly here and eventually allow for other formats. Currently, yaml only is offered since this is what we accept in.